src/api/api_core.php

```diff
@@ -54,13 +54,9 @@ function login($username, $password) {
             $query->execute(array(':id'       => $result['id'],
                                   ':auth_key' => $auth_key));
 
-            // set authorisation cookie
-            setcookie('auth_userid', $result['id'], 0, '/', '', FALSE);
-            setcookie('auth_key', $auth_key, 0, '/', '', FALSE);
-            session_regenerate_id(TRUE);
-            $_SESSION['user_id'] = $result['id'];
-            $_SESSION['user_name'] = $result['name_ingame'];
-            $_SESSION['user_lastactive'] = time();
+            update_session($result['id'], $result['name_ingame']);
+            set_authorisation_cookies($result['id'], $auth_key, $doStayLoggedIn);
```

Comment:
Responding to my previous comment: the obvious answer here is to ask, since our goal is to keep the session and the cookies in sync (i think, anyway), is either of updating the session or setting the authorization cookies likely to fail a fraction of the time we care about at all?  If so, that one should happen first.